File: java-to-go-projects/10-tcp-chat-server/Message.java
package chat;

import java.time.Instant;

/**
 * Represents a single chat message with sender, room, content and timestamp.
 */
public class Message {
    private final String sender;
    private final String room;
    private final String content;
    private final Instant timestamp;

    public Message(String sender, String room, String content) {
        this.sender = sender;
        this.room = room;
        this.content = content;
        this.timestamp = Instant.now();
    }

    public String getSender() { return sender; }
    public String getRoom() { return room; }
    public String getContent() { return content; }
    public Instant getTimestamp() { return timestamp; }

    /**
     * Format for display: [room] sender: content
     */
    public String format() {
        return String.format("[%s] %s: %s", room, sender, content);
    }
}

File: java-to-go-projects/10-tcp-chat-server/rust/src/message.rs
use std::fmt;

pub struct Message {
    pub sender: String,
    pub room: String,
    pub content: String,
}

impl Message {
    pub fn new(sender: &str, room: &str, content: &str) -> Self {
        Message {
            sender: sender.to_string(),
            room: room.to_string(),
            content: content.to_string(),
        }
    }
}

impl fmt::Display for Message {
    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
        write!(f, "[{}] {}: {}", self.room, self.sender, self.content)
    }
}
